src/main/java/com/doctusoft/math/ClosedRange.java

```diff
@@ -125,19 +253,173 @@ public boolean isValidUpperBound(C upperBound) {
+        int result = lowerBound.hashCode();
+        result = 31 * result + upperBound.hashCode();
+        return result;
+    }
+    
+    /**
+     * Important to note that the {@link ClosedRange#toString()} representation follows the mathematical interval
+     * notation of {@code [lowerBound; upperBound]}. Using the "; " as separator is important since some number formats
+     * may use the ',' in their {@link String} representation. The type of the interval's type argument is responsible
+     * for providing its own {@link String} representation with its {@code toString()} method.
+     */
     public String toString() {
-        return "[" + lowerBound + ", " + upperBound + "]";
+        return new StringBuilder().append(OPEN_SYMBOL).append(lowerBound + SEPARATOR + upperBound).append(CLOSE_SYMBOL).toString();
     }
-
-    public static int intSizeExact(ClosedRange<Integer> intRange) {
-        long length = intRange.getUpperBound().longValue() - intRange.getLowerBound().longValue();
-        return Math.toIntExact(length + 1L);
+    
+    private static final char OPEN_SYMBOL = '[';
+    
+    private static final String SEPARATOR = "; ";
+    
+    private static final char CLOSE_SYMBOL = ']';
+    
+    @Beta
+    public static final BigDecimal decimalLengthOf(ClosedRange<BigDecimal> decimalRange) {
+        return decimalRange.getUpperBound()
+            .subtract(decimalRange.getLowerBound());
     }
-
-    public static long longSizeExact(ClosedRange<Long> longRange) {
-        BigInteger length = BigInteger.valueOf(longRange.getUpperBound())
-            .subtract(BigInteger.valueOf(longRange.getLowerBound()));
-        return length.add(BigInteger.ONE).longValueExact();
+    
+    /**
+     * Computes the discrete element count of the provided {@code ClosedRange<BigInteger>} interval.
+     */
+    public static final BigInteger countBigints(ClosedRange<BigInteger> bigintRange) {
+        return bigintRange.getUpperBound()
+            .subtract(bigintRange.getLowerBound())
+            .add(BigInteger.ONE);
     }
-
+    
+    /**
+     * Computes the discrete element count of the provided {@code ClosedRange<Long>} interval. The return value is a
+     * {@link BigInteger}, because the element count for valid interval could be larger than {@link Long#MAX_VALUE},
+     * however the {@link BigInteger} class has a wide range of methods to do the necessary (checked) conversions
+     * where needed.
+     */
+    public static final BigInteger countLongs(ClosedRange<Long> longRange) {
+        return countBigints(longRange.convert(BigInteger::valueOf));
+    }
+    
+    /**
+     * Computes the discrete element count of the provided {@code ClosedRange<Integer>} interval. The return value is a
+     * {@link BigInteger}, because the element count for valid interval could be larger than {@link Integer#MAX_VALUE},
+     * however the {@link BigInteger} class has a wide range of methods to do the necessary (checked) conversions
```

Comment:
The however is not good here you could use furthermore instead.
You can also get rid of the next sentence with this change.
The same sentence is present in the LeftClosedRange class.